opslevel/resource_opslevel_service_tool.go

```diff
@@ -182,6 +182,10 @@ func (r *ServiceToolResource) Read(ctx context.Context, req resource.ReadRequest
 		return
 	}
 
+	if err := service.Hydrate(r.client); err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to hydrate service (%s), got error: %s", service.Id, err))
```

Comment:
This `Read()` function should stop executing at this point if there is an error.